src/GraphQL.MicrosoftDI/ResolverBuilder.cs

```diff
@@ -0,0 +1,313 @@
+        public FieldBuilder<TSourceType, TReturnType> ResolveAsync(Func<IResolveFieldContext<TSourceType>, T1, T2, T3, T4, Task<TReturnType>> resolver)
+        {
+            Func<IResolveFieldContext<TSourceType>, Task<TReturnType>> resolver2 =
+                (context) => resolver(
+                    context,
+                    context.RequestServices.GetRequiredService<T1>(),
+                    context.RequestServices.GetRequiredService<T2>(),
+                    context.RequestServices.GetRequiredService<T3>(),
+                    context.RequestServices.GetRequiredService<T4>());
+
+            return _scoped ? _builder.ResolveScopedAsync(resolver2) : _builder.ResolveAsync(resolver2);
+        }
+    }
+
+    /// <summary>
+    /// A builder for field resolvers with 5 extra service types.
+    /// </summary>
+    public class ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5>
+    {
+        private readonly FieldBuilder<TSourceType, TReturnType> _builder;
+        private bool _scoped;
+
+        /// <inheritdoc cref="ResolverBuilder{TSourceType, TReturnType}.ResolverBuilder(FieldBuilder{TSourceType, TReturnType}, bool)"/>
+        public ResolverBuilder(FieldBuilder<TSourceType, TReturnType> builder, bool scoped)
+        {
+            _builder = builder;
+            _scoped = scoped;
+        }
+
+        /// <inheritdoc cref="ResolverBuilder{TSourceType, TReturnType}.WithScope"/>
+        public ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5> WithScope()
+        {
+            _scoped = true;
+            return this;
+        }
+
+        /// <inheritdoc cref="ResolverBuilder{TSourceType, TReturnType}.Resolve(Func{IResolveFieldContext{TSourceType}, TReturnType})"/>
+        public FieldBuilder<TSourceType, TReturnType> Resolve(Func<IResolveFieldContext<TSourceType>, T1, T2, T3, T4, T5, TReturnType> resolver)
+        {
+            Func<IResolveFieldContext<TSourceType>, TReturnType> resolver2 =
+                (context) => resolver(
+                    context,
+                    context.RequestServices.GetRequiredService<T1>(),
+                    context.RequestServices.GetRequiredService<T2>(),
+                    context.RequestServices.GetRequiredService<T3>(),
+                    context.RequestServices.GetRequiredService<T4>(),
+                    context.RequestServices.GetRequiredService<T5>());
+
+            return _scoped ? _builder.ResolveScoped(resolver2) : _builder.Resolve(resolver2);
+        }
+
+        /// <inheritdoc cref="ResolverBuilder{TSourceType, TReturnType}.ResolveAsync(Func{IResolveFieldContext{TSourceType}, Task{TReturnType}})"/>
+        public FieldBuilder<TSourceType, TReturnType> ResolveAsync(Func<IResolveFieldContext<TSourceType>, T1, T2, T3, T4, T5, Task<TReturnType>> resolver)
+        {
+            Func<IResolveFieldContext<TSourceType>, Task<TReturnType>> resolver2 =
+                (context) => resolver(
```

Comment:
```suggestion
                context => resolver(
```

Reply:
Personally I think `(context) => ....` is more explicit; it shows that `context` is an argument better; that's why I always do it that way.